EventFilter/Phase2PixelRawToDigi/plugins/Phase2ITQCoreProducer.cc

```diff
@@ -0,0 +1,226 @@
+    */
+  std::for_each(hitList.begin(), hitList.end(), &updateHitCoordinatesForLargePixels);
+  return hitList;
+}
+
+std::vector<Phase2ITChip> splitByChip(std::vector<Phase2ITDigiHit> hitList) {
+  // Split the hit list by read out chip
+  std::vector<std::vector<Phase2ITDigiHit>> hits_per_chip(4);
+  for (auto hit : hitList) {
+    int chip_index = (hit.col() < kQCoresInChipColumn) ? 0 : 1;
+    if (hit.row() >= kQCoresInChipRow) {
+      chip_index += 2;
+    }
+    hits_per_chip[chip_index].push_back(hit);
+  }
+
+  // Generate Phase2ITChip objects from the hit lists
+  std::vector<Phase2ITChip> chips;
+  for (int chip_index = 0; chip_index < 4; chip_index++) {
+    chips.push_back(Phase2ITChip(chip_index, hits_per_chip[chip_index]));
+  }
+
+  return chips;
+}
+
+std::vector<Phase2ITChip> processHits(std::vector<Phase2ITDigiHit> hitList) {
+  std::vector<Phase2ITDigiHit> newHitList;
+  newHitList = adjustEdges(hitList);
+
+  std::vector<Phase2ITChip> chips = splitByChip(newHitList);
+  std::vector<bool> code;
+
+  for (size_t i = 0; i < chips.size(); i++) {
+    Phase2ITChip chip = chips[i];
+    code = chip.get_chip_code();
+  }
+
+  return chips;
+}
+
+// ------------ method called to produce the data  ------------
+void Phase2ITQCoreProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace std;
+
+  unique_ptr<edm::DetSetVector<Phase2ITQCore>> aQCoreVector = make_unique<edm::DetSetVector<Phase2ITQCore>>();
+  unique_ptr<edm::DetSetVector<Phase2ITChipBitStream>> aBitStreamVector =
+      make_unique<edm::DetSetVector<Phase2ITChipBitStream>>();
+
+  auto const& tTopo = iSetup.getData(tTopoToken_);
+
+  edm::Handle<edm::DetSetVector<PixelDigi>> pixelDigiHandle;
+  iEvent.getByToken(pixelDigi_token_, pixelDigiHandle);
+
+  edm::DetSetVector<PixelDigi>::const_iterator iterDet;
+  for (iterDet = pixelDigiHandle->begin(); iterDet != pixelDigiHandle->end(); iterDet++) {
+    DetId tkId = iterDet->id;
+    edm::DetSet<PixelDigi> theDigis = (*pixelDigiHandle)[tkId];
+    std::vector<Phase2ITDigiHit> hitlist;
+    std::vector<int> id;
+
+    if (tkId.subdetId() == PixelSubdetector::PixelBarrel) {
+      int layer_num = tTopo.pxbLayer(tkId.rawId());
+      int ladder_num = tTopo.pxbLadder(tkId.rawId());
+      int module_num = tTopo.pxbModule(tkId.rawId());
+      id = {tkId.subdetId(), layer_num, ladder_num, module_num};
+    } else if (tkId.subdetId() == PixelSubdetector::PixelEndcap) {
+      int module_num = tTopo.pxfModule(tkId());
+      int disk_num = tTopo.pxfDisk(tkId());
+      int blade_num = tTopo.pxfBlade(tkId());
+      int panel_num = tTopo.pxfPanel(tkId());
+      int side_num = tTopo.pxfSide(tkId());
+      id = {tkId.subdetId(), module_num, disk_num, blade_num, panel_num, side_num};
+    }
+
+    for (auto iterDigi = theDigis.begin(); iterDigi != theDigis.end(); ++iterDigi) {
+      hitlist.emplace_back(Phase2ITDigiHit(iterDigi->row(), iterDigi->column(), iterDigi->adc()));
```

Comment:
Could be range-for, and given `emplace_back` the explicit constructor call can be omitted
```suggestion
    for (const auto& digi : theDigis) {
      hitlist.emplace_back(digi.row(), digi.column(), digi.adc());
```